### PR TITLE
fix: send last error response for aborted jobs to reporting

### PR DESCRIPTION
**Pull request by cisse21**
# Description

This PR sends the last error response to reporting and in the job status table for the jobs which get aborted due the exhausting their retries

## Linear Ticket

[ Linear Link ](https://linear.app/rudderstack/issue/PIPE-78/propagate-error-response-for-aborted-jobs-to-reporting)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 ## Notion Ticket
 
-< Replace with Notion Link >
```
Comment (atzoum) [inline .github/pull_request_template.md:7]: please create a separate PR for this kind of changes

File: router/worker.go [hunk 2]
```diff
@@ -88,12 +89,14 @@ func (w *worker) workLoop() {
 				panic(fmt.Errorf("unmarshalling of job parameters failed for job %d (%s): %w", job.JobID, string(job.Parameters), err))
 			}
 			w.rt.destinationsMapMu.RLock()
+			var abortTag string
```
Comment (atzoum) [inline router/worker.go:92]:
```suggestion
```

File: router/worker.go [hunk 3]
```diff
@@ -88,12 +89,14 @@ func (w *worker) workLoop() {
...
 			abort, abortReason := routerutils.ToBeDrained(job, parameters.DestinationID, w.rt.reloadableConfig.toAbortDestinationIDs, w.rt.destinationsMap)
+			abortTag = abortReason
```
Comment (atzoum) [inline router/worker.go:94]:
```suggestion
			abortTag := abortReason
```

File: router/worker.go [hunk 4]
```diff
@@ -88,12 +89,14 @@ func (w *worker) workLoop() {
...
 			w.rt.destinationsMapMu.RUnlock()
-
 			if !abort {
 				abort = w.retryLimitReached(&job.LastJobStatus)
-				abortReason = "retry limit reached"
+				abortReason = string(job.LastJobStatus.ErrorResponse)
+				abortTag = "retry_limit_reached"
```
Comment (atzoum) [inline router/worker.go:99]: Why do we need underscores?
```suggestion
				abortTag = "retry limit reached"
```

**Review by atzoum — APPROVED** [review]

**Review by Sidddddarth — APPROVED** [review]

**Pull request merged**